src/main/java/org/reaktivity/nukleus/oauth/internal/OAuthRealms.java

```diff
@@ -121,7 +130,11 @@ public boolean unresolve(
         long authorization)
     {
         final long realmId = authorization & REALM_MASK;
-        return Long.bitCount(realmId) <= 1 && realmsByName.entrySet().removeIf(e -> e.getValue().realmId == realmId);
+        final OAuthRealm realm = realmsByName.values().stream()
+                                                         .filter(rs -> rs.unresolve(realmId))
+                                                         .findFirst().orElse(null);
+        realmsByName.values().removeIf(OAuthRealm::isEmpty);
```

Comment:
Hoist `realmsByName.values()` into `realms` variable as it is used twice.
Excellent use of the `OAuthRealm::isEmpty` method reference!